core/src/main/java/tech/tablesaw/api/Row.java

```diff
@@ -9,10 +9,11 @@
 import java.util.Map;
 import java.util.TreeMap;
 import tech.tablesaw.columns.Column;
+import tech.tablesaw.table.TableSlice;
 
 public class Row implements Iterator<Row> {
 
-  private final Table table;
+  private final TableSlice tableSlice;
```

Comment:
Maybe we should use `Relation` instead?

Reply:
We could use relation, but the obvious implementation would require adding yet another method to Table. It would be nice to avoid that if we can.

To iterate over a Selection of Rows we need to map the row number in the view to the row number on the underlying table. This PR creates a method on TableSlice to do this called `mappedRowNumber`. If we were to use Relation (and a similar implementation) it would  involve adding another method to Relation and overriding it in TableSlice.